src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -1471,8 +1477,15 @@ window.qBittorrent.DynamicTable = (function() {
             const filteredRows = [];
 
             const rows = this.rows.getValues();
+            const torrentsFilterRegex = $('torrentsFilterRegexBox').checked;
             const filterText = $('torrentsFilterInput').value.trim().toLowerCase();
-            const filterTerms = (filterText.length > 0) ? filterText.split(" ") : null;
+            let filterTerms;
+            if (torrentsFilterRegex) {
+                filterTerms = (filterText.length > 0) ? new RegExp(filterText) : null;
+            }
+            else {
+                filterTerms = (filterText.length > 0) ? filterText.split(" ") : null;
+            }
```

Comment:
```suggestion
            const filterTerms = (filterText.length > 0)
                ? (useRegex ? new RegExp(filterText) : filterText.split(" "))
                : null;
```
ps. not sure this will pass the formatting check, run `npm run format` to format it.